src/lib.rs

```diff
@@ -349,3 +349,69 @@ enum DerInner<'a> {
     Owned(Vec<u8>),
     Borrowed(&'a [u8]),
 }
+
+/// An encoding of the PKIX AlgorithmIdentifier type:
+///
+/// ```ASN.1
+/// AlgorithmIdentifier  ::=  SEQUENCE  {
+///     algorithm               OBJECT IDENTIFIER,
+///     parameters              ANY DEFINED BY algorithm OPTIONAL  }
+///                                -- contains a value of the type
+///                                -- registered for use with the
+///                                -- algorithm object identifier value
+/// ```
+/// (from <https://www.rfc-editor.org/rfc/rfc5280#section-4.1.1.2>)
+///
+/// The outer sequence encoding is *not included*, so this is an OID encoding
+/// for `algorithm` plus the `parameters` value.
+///
+/// For example, this is the `rsaEncryption` algorithm:
+///
+/// ```
+/// let rsa_encryption = rustls_pki_types::AlgorithmIdentifier::from_slice(
+///     &[
+///         // algorithm: 1.2.840.113549.1.1.1
+///         0x06, 0x09, 0x2a, 0x86, 0x48, 0x86, 0xf7, 0x0d, 0x01, 0x01, 0x01,
+///         // parameters: NULL
+///         0x05, 0x00
+///     ]
+/// );
+/// ```
+#[derive(Debug, Clone, PartialEq)]
+pub struct AlgorithmIdentifier<'a>(Der<'a>);
```

Comment:
Should we rename this to `SignatureAlgorithmIdentifier` instead?

Reply:
Since it parameters in addition to OID I think that makes sense.

Reply:
i'm minded to keep `AlgorithmIdentifier` to match the naming in RFC5280. plus -- annoyingly -- there's not a separate namespace of these used in signatures and public keys.